### ISSUE TITLE
Array.map3 & tuples producing incorrect results

### ISSUE THREAD

**Issue by pkese** [fable-compiler/Fable#2030]
I've noticed that `Array.map3` produces wrong results when returning tuples.

Here is an example:
```fsharp
    let aa = Array.init 10 id
    let tuples = (aa,aa,aa) |||> Array.map3 (fun a b c -> a,Some b,c)
    //printfn "tuples: %A" tuples
    let aa' = tuples |> Array.map (fun (a,b,c) -> (a+b.Value+c)/3)
    //printfn "aa': %A" aa'
    assert (aa = aa')
```

Similar code but with Array.map2 works fine:
```fsharp
    let aa = Array.init 10 id
    let tuples = (aa,aa) ||> Array.map2 (fun a b -> a,Some b)
    let aa' = tuples |> Array.map (fun (a,b) -> (a+b.Value)/2)
    assert (aa = aa')
```

This is on fable-compiler 2.8.0

**Comment by pkese** [comment]
... the above code produces:
```
tuples: 0,0,0,0,0,0,0,0,0,0
aa': 0,0,0,0,0,0,0,0,0,0
```
or with `console.log`
```javascript
tuples = Int32Array(10) [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ]
aa = Int32Array(10) [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ]
```

Instead, if it was correct, that should be:
```fsharp
tuples = [| (0,Some 0,0); (1, Some 1,1), (2, Some 2,2),  ... , (9, Some 9,9)]
aa' = [| 0; 1; 2; 3; 4; 5; 6; 7; 8; 9 |]
```

**Comment by MangelMaxime** [comment]
It has been released in Fable 2.8.1
